backend/app/controllers/reading_lists_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class ReadingListsController < ApplicationController
+  before_action :authorize_request
+
+  def create
+    if reading_list.valid?
+      render(json: { data: reading_list }, status: :created)
+    else
+      render(json: { errors: reading_list.errors.full_messages }, status: :unprocessable_entity)
+    end
+  end
+
+  def update
+    if !user_owns_reading_list?
+      render(json: { errors: ["Unauthorized request detected"] }, status: :unauthorized)
```

Comment:
Is it really unauthorized? I think this falls more in `not found` error category. Because the resource `is not found` within the bounds of the parent user 🤔

Reply:
Hmmm. The record is there, so I'm checking if it belongs to the user. The record is being checked if `reading_list.user_id` is owned by the current user accessing the method.

Reply:
Oh, got it from the comment below. If I changed it to that logic, makes sense for `not found`